ibis/backends/dask/execution/util.py

```diff
@@ -37,3 +39,55 @@ def make_selected_obj(gs: SeriesGroupBy):
         return gs.obj.set_index(gs.index, drop=False)[
             gs._meta._selected_obj.name
         ]
+
+
+def maybe_wrap_scalar(result: Any, expr: ir.Expr) -> Any:
+    """
+    A partial implementation of `coerce_to_output` in the pandas backend.
+
+    Currently only wraps scalars, but will change when udfs are added to the
+    dask backend.
+    """
+    result_name = expr.get_name()
+    if isinstance(result, dd.core.Scalar) and isinstance(
+        expr.op(), ops.Reduction
+    ):
+        # TODO - computation
+        return dd.from_pandas(
+            pd.Series(result.compute(), name=result_name), npartitions=1
+        )
+        # return dd.Series(result, name=result_name)
+    else:
+        return result.rename(result_name)
+
+
+def safe_concat(dfs: List[Union[dd.Series, dd.DataFrame]]) -> dd.DataFrame:
+    """
+    Concat a list of `dd.Series` or `dd.DataFrame` objects into one DataFrame
+
+    This will use `DataFrame.concat` if all pieces are the same length.
+    Otherwise we will iterratively join.
+
+    When axis=1 and divisions are unknown, Dask `DataFrame.concat` can only
+    operate on objects with equal lengths, otherwise it will raise a
+    ValueError in `concat_and_check`.
+
+    See https://github.com/dask/dask/blob/2c2e837674895cafdb0612be81250ef2657d947e/dask/dataframe/multi.py#L907 # noqa
+
+    Note - this is likely to be quite slow, but this should be hit rarely in
+    real usage. A situtation that triggeres this slow path is aggregations
+    where aggregations return different numbers of rows (see
+    `test_aggregation_group_by` for a specific example).
```

Comment:
Can you be more specific what the case here is? I took a look at https://github.com/ibis-project/ibis/blob/17628efdf12a2ff2f4192c4ce04dff5a86608483/ibis/backends/dask/tests/execution/test_operations.py#L118 but didn't know case do mean.

Reply:
Any of the cases hit this issue. The problem is that your have an aggregation that returns different numbers of rows depend on whether it is masked or not. 

https://github.com/ibis-project/ibis/blob/17628efdf12a2ff2f4192c4ce04dff5a86608483/ibis/backends/dask/tests/execution/test_operations.py#L126-L127

Reply:
Sorry I think I am missing sth here. I thought the aggregation returns n rows where n is the number of groups?

Reply:
`neg_mean_int64_with_zeros=(-t.int64_with_zeros).mean(where=ibis_where),` will only return rows (and perform the calculation) where the mask is true. 

`nunique_dup_ints` will run for each group. 

So if you're masking by your grouping column those two operations will return different numbers of rows.